src/Components/Groups/Group.tsx

```diff
@@ -0,0 +1,289 @@
+import GroupsDropdown from "./GroupsDropdown";
+import { useTranslation } from "react-i18next";
+
+const Group: React.FC<IGroupProps> = ({
+  id,
+  name,
+  setGroupList,
+  groupList,
+  currentRole,
+}) => {
+  const [isNameEditable, setIsNameEditable] = useState(false);
+  const [isPeopleOpen, setIsPeopleOpen] = useState(false);
+  const [isGroupsOpen, setIsGroupsOpen] = useState(false);
+  const [people, setPeople] = useState<PersonWithId[]>([]);
+  const [groups, setGroups] = useState<IGroup[]>([]);
+  const [groupName, setGroupName] = useState(name);
+  const [allPeople, setAllPeople] = useState<JSX.Element[]>([]);
+  const [allGroups, setAllGroups] = useState<JSX.Element[]>([]);
+  const [selectedIndex, setSelectedIndex] = useState<number>();
+  const [selectedGroupIndex, setSelectedGroupIndex] = useState<number>();
+
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    getGroups();
+  }, [groupList]);
+
+  const deleteGroup = () => {
+    GroupServices.deleteGroup(id)
+      .then(() => {
+        setGroupList((groupList: IGroup[]) =>
+          groupList.filter((group) => group.id !== id)
+        );
+        return toast.success(t("group.delete"));
+      })
+      .catch((err) => {
+        return toast.error(err.response.data);
+      });
+  };
+
+  const updateName = () => {
+    GroupServices.updateName(id, groupName)
+      .then(() => {
+        setGroupList((groupList: IGroup[]) =>
+          groupList.map((group: IGroup) => {
+            if (group.id === id) return { id, name: groupName };
+            return { id: group.id, name: group.name };
+          })
+        );
+        return toast.success(t("group.update"));
+      })
+      .catch((err) => {
+        return toast.error(err.response.data);
+      });
+  };
+
+  useEffect(() => {
+    PeopleServices.getAllPeople()
+      .then((res) => {
+        const peopleId = people.map(({ id }) => id);
+        setAllPeople(
+          res
+            .filter((person: PersonWithId) => !peopleId.includes(person.id))
+            .map((person: IAutocomplete, index: number) => {
+              return (
+                <ListItemButton
+                  key={person.id}
+                  selected={selectedIndex === index}
+                  onClick={(event) => {
+                    handleListItemClick(event, 1);
+                    addPerson(person.id);
+                  }}
+                >
+                  <ListItemText primary={person.name} />
+                </ListItemButton>
+              );
+            })
+        );
+      })
+      .catch((err) => {
+        return toast.error(err.response.data);
+      });
+  }, [people]);
+
+  useEffect(() => {
+    GroupServices.getAllGroups()
+      .then((res) => {
+        const groupId = groups.map((group: IGroup) => {
+          return group.id;
+        });
+        setAllGroups(
+          res
+            .filter(
+              (group: IGroup) => !groupId.includes(group.id) && id !== group.id
```

Comment:
Flip the statement to ascend in complexity, thus first executing less time-consuming checks. 
```suggestion
              (group: IGroup) => id !== group.id && !groupId.includes(group.id) 
```